### ISSUE TITLE
EXPO - EAS build failing 

### ISSUE THREAD

**Issue by yonitou** [nrwl/nx#13242]
### Current Behavior

Hi, 
I'm currently migrating two repos to an NX mono repository (React.JS + Expo). 

Everything was going well until yesterday. All my EAS builds are failing and I can't understand why (they worked perfectly fine when I used a standalone repo)

### Expected Behavior

The EAS build should be succeeded when using @nrwl/expo:build. 

### Steps to Reproduce

https://github.com/yonitou/example

1. Go to the GitHub repo above
2. Run npm install
3. nx build native-app --configuration=simulator
4. Wait 

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.13.1
   OS   : darwin arm64
   npm  : 8.3.0
   
   nx : 15.0.13
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.0.13
   @nrwl/detox : 15.1.1
   @nrwl/devkit : 15.0.13
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.13
   @nrwl/expo : 15.1.1
   @nrwl/express : Not Found
   @nrwl/jest : 15.1.1
   @nrwl/js : 15.0.13
   @nrwl/linter : 15.0.13
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.0.13
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.1.1
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.13
   @nrwl/web : 15.1.1
   @nrwl/webpack : 15.1.1
   @nrwl/workspace : 15.0.13
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```
```


### Failure Logs

```shell
The failure logs provided by EXPO are : 


 FAILURE: Build failed with an exception.
49
[stderr] * Where:
50
[stderr] Script '/home/expo/workingdir/build/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' line: 250
51
[stderr] * What went wrong:
52
[stderr] A problem occurred evaluating script.
53
[stderr] > Calling `[node, /home/expo/workingdir/build/node_modules/@react-native-community/cli/build/bin.js, config]` finished with an exception. Error message: groovy.json.JsonException: Unable to determine the current character, it is not a string, number, array, or object
54
[stderr]   
55
[stderr]   The current character read is 'i' with an int value of 105
56
[stderr]   Unable to determine the current character, it is not a string, number, array, or object
57
[stderr]   line number 1
58
[stderr]   index number 0
59
[stderr]   info Run CLI with --verbose flag for more details.
60
[stderr]   ^. Output: info Run CLI with --verbose flag for more details.
```
```


### Additional Information

_No response_

**Comment by AgentEnder** [comment]
Hey @yonitou, it may be unrelated, but some of your @nrwl/* packages don't line up with the version of `nx` you have installed. Aside from `@nrwl/nx-cloud`, all packages under the nrwl scope should generally be set up with the same version. 

Can you correct this and check that the issue still reproduces? An easy way to fix this would be to run `nx migrate {v}`, where `v` is the version of nx you have installed. Alternatively, updating to latest with `nx migrate latest` would also work.

**Comment by yonitou** [comment]
Hey ! Thanks for your help. 

I just ran nx migrate latest. It worked without any errors but it doesn't change anything. It really feels like my package-lock.json is kind of corrupted but I don't think I should delete it.. I'm running out of ideas, did you have a look on my repository that I shared ?

**Comment by yonitou** [comment]
I think I just found one possible answers by mistake => The builds are failing when `requireCommit` is set to true in eas.json file. I just set it to false and it's not failing ! The huge difference is also that when this param is set to true, my app bundle is 23 MB. When it's set to false, it has its "normal size" of 9MB. There is something reaaaally buggy and weird right here so I'm still looking for some explanation :)

**Comment by yonitou** [comment]
I just found the issue and this is honestly one of the weirdest bugs ever => It's because I deleted `expo-status-bar` from my package.json (because I don't use it). It's a package that comes automatically with the NX generators when generating an expo app. For some reason, it's breaking the build process (with the error logs above) if it's not present. So for now, I'm curious about why -I didn't find anything interesting when looking at your codebase- ..
